iothub_client/samples/iothub_client_device_cbor_sample/iothub_client_device_cbor_sample.c

```diff
@@ -0,0 +1,956 @@
+        if (rc)
+        {
+            (void)printf("Failed to create %s map: CborError %d.\n", twin_property_state_name, rc);
+            exit(rc);
+        }
+
+            // state.max_speed
+            rc = cbor_encode_text_string(&state_map, twin_property_state_max_speed_name, strlen(twin_property_state_max_speed_name));
+            if (rc)
+            {
+                (void)printf("Failed to encode text string '%s': CborError %d.\n", twin_property_state_max_speed_name, rc);
+                exit(rc);
+            }
+            rc = cbor_encode_uint(&state_map, device_car.state.max_speed);
+            if (rc)
+            {
+                (void)printf("Failed to encode int '%" PRIu64 "': CborError %d.\n", device_car.state.max_speed, rc);
+                exit(rc);
+            }
+
+            // state.software_version
+            rc = cbor_encode_text_string(&state_map, twin_property_state_software_version_name, strlen(twin_property_state_software_version_name));
+            if (rc)
+            {
+                (void)printf("Failed to encode text string '%s': CborError %d.\n", twin_property_state_software_version_name, rc);
+                exit(rc);
+            }
+            rc = cbor_encode_double(&state_map, device_car.state.software_version);
+            if (rc)
+            {
+                (void)printf("Failed to encode double '%.*f': CborError %d.\n", 1, device_car.state.software_version, rc);
+                exit(rc);
+            }
+
+            // state.vanity_plate
+            rc = cbor_encode_text_string(&state_map, twin_property_state_vanity_plate_name, strlen(twin_property_state_vanity_plate_name));
+            if (rc)
+            {
+                (void)printf("Failed to encode text string '%s': CborError %d.\n", twin_property_state_vanity_plate_name, rc);
+                exit(rc);
+            }
+            rc = cbor_encode_text_string(&state_map, device_car.state.vanity_plate, strlen(device_car.state.vanity_plate));
+            if (rc)
+            {
+                (void)printf("Failed to encode text string '%s': CborError %d.\n",  device_car.state.vanity_plate, rc);
+                exit(rc);
+            }
+
+        rc = cbor_encoder_close_container(&encoder_map, &state_map);
+        if (rc)
+        {
+            (void)printf("Failed to close %s container: CborError %d.\n", twin_property_state_name, rc);
+            exit(rc);
+        }
+
+    rc = cbor_encoder_close_container(&encoder, &encoder_map);
+    if (rc)
+    {
+        (void)printf("Failed to close container: CborError %d.\n", rc);
+        exit(rc);
+    }
+
+    *out_reported_property_payload_length = cbor_encoder_get_buffer_size(&encoder, reported_property_payload);
+}
+
+static void build_cbor_telemetry(uint8_t* telemetry_payload, size_t telemetry_payload_size, size_t* out_telemetry_payload_length)
+{
+  float engine_temperature = 200.0f + ((float)rand() / RAND_MAX) * 5.0f;
```

Comment:
correct me if I'm wrong here but 
```c
((float)rand() / RAND_MAX)
```
would return `0` every time correct? [`rand()` is between `0` and `RAND_MAX`](https://www.tutorialspoint.com/c_standard_library/c_function_rand.htm) so dividing by it would be 0.

maybe
```suggestion
  #define TEMPERATURE_VARIANCE 50
  float engine_temperature = 200.0f + ((float)rand() % TEMPERATURE_VARIANCE) * 5.0f;
```

Reply:
no.  rand() returns an integer but is cast to a float.  a float divided by an integer returns a float.